### PR TITLE
Vulkan: Set object name and tag.

### PR DESCRIPTION
**Pull request by Qining**

### CODE REVIEW

**Comment by Qining** [comment]
In vulkan_mid_execution.cpp, the real changes are:
1) add function : `recreateDebugInfo`
2) add function calls to `recreateDebugInfo` after each recreated Vulkan handle.
All other changes in that file are done by clang-format

In vulkan.api, the reason to have two `switch..case` list is that, I cannot get the reference of the DebugInfo pointer in objects in API file.

File: gapii/cc/vulkan_mid_execution.cpp [hunk 1]
```diff
@@ -153,1690 +156,1780 @@ class TemporaryRenderPass {
...
+                        }) != temporary_render_passes_.end();
   }
 
  private:
   CallObserver* observer_;
   VulkanSpy* spy_;
   std::vector<std::shared_ptr<RenderPassObject>> temporary_render_passes_;
 };
+
+template <typename ObjectClass>
```
Comment (Qining) [inline gapii/cc/vulkan_mid_execution.cpp]: Real change starts from here.

File: gapii/cc/vulkan_mid_execution.cpp [hunk 2]
```diff
@@ -153,1690 +156,1780 @@ class TemporaryRenderPass {
...
+      v.mpQueuePriorities = queue_priorities[i].data();
+    }
+    create_info.mqueueCreateInfoCount = queue_create_infos.size();
+    create_info.mpQueueCreateInfos = queue_create_infos.data();
+    create_info.mpEnabledFeatures = &device.second->mEnabledFeatures;
+
+    VkDevice d = device.second->mVulkanHandle;
+    VkPhysicalDevice pd = device.second->mPhysicalDevice;
+    RecreateDevice(observer, pd, &create_info, &d);
+    recreateDebugInfo(
```
Comment (Qining) [inline gapii/cc/vulkan_mid_execution.cpp]: For each recreate vulkan handle, createDebugInfo() is called, although I don't think debug info can be attached to VkInstance, VkPhysicalDevice, VkSurface.

File: gapis/api/vulkan/vulkan.api [hunk 3]
```diff
@@ -8700,6 +8700,229 @@ class VkDebugMarkerMarkerInfoEXT {
...
+	  	Surfaces[as!VkSurfaceKHR(obj)].DebugInfo = updateDebugMarkerObjectName(
+          Surfaces[as!VkSurfaceKHR(obj)].DebugInfo, nameInfo)
+	  case VK_DEBUG_REPORT_OBJECT_TYPE_SWAPCHAIN_KHR_EXT:
+	  	Swapchains[as!VkSwapchainKHR(obj)].DebugInfo = updateDebugMarkerObjectName(
+          Swapchains[as!VkSwapchainKHR(obj)].DebugInfo, nameInfo)
+  }
+}
+
+@override
+@no_replay
```
Comment (Qining) [inline gapis/api/vulkan/vulkan.api]: Will be changed later to 'custom' once we start to use the debug markers for grouping

File: gapis/api/vulkan/vulkan.api [hunk 4]
```diff
@@ -8709,10 +8932,20 @@ class VkDebugMarkerMarkerInfoEXT {
 cmd VkResult vkDebugMarkerSetObjectTagEXT(
```
Comment (Qining) [inline gapis/api/vulkan/vulkan.api]: Also will be changed to 'custom' in later CLs

File: gapii/cc/vulkan_mid_execution.cpp [hunk 5]
```diff
@@ -153,1690 +156,1780 @@ class TemporaryRenderPass {
...
+
+template <>
+VkDevice getObjectCreatingDevice<SurfaceObject>(
+    std::shared_ptr<SurfaceObject> obj) {
+  return VkDevice(0);
+}
+
+template <typename ObjectClass>
+void recreateDebugInfo(VulkanSpy* spy, CallObserver* observer,
+                       uint32_t objectType, std::shared_ptr<ObjectClass> obj) {
```
Comment (AWoloszyn) [inline gapii/cc/vulkan_mid_execution.cpp]: You should probably make this a const std::shared_ptr<>& obj, unless you need to modify the object.
shared_ptr copies are more expensive than we want to pay if we don't have to.

Comment (Qining) [reply gapii/cc/vulkan_mid_execution.cpp]: Changed to const ref in a83cc08

**Pull request merged**